lib/eventTarget.js

```diff
@@ -0,0 +1,149 @@
+   * @method preventDefault
+   */
+  preventDefault: function() {
+    this.defaultPrevented = true;
+  }
+};
+
+EventTarget.prototype = {
+  /**
+   * Fired before the default function has occurred.
+   * Call e.preventDefault() on the event and the action we be prevented.
+   * @method on
+   * @param  {String} topic event to listen to
+   * @param  {Function} callback
+   * @param  {Object} [context]
+   * @param  [...more args...] can be provided and will be passed to the callback}
+   */
+  on: function(topic, callback, context /*adtl args*/) {
+    return this._registerEvent('on', topic, callback, context, Array.prototype.slice.call(arguments, 3));
+  },
+
+  /**
+   * Fired after the default function has occurred.
+   * @method after
+   * @param  {String} topic event to listen to
+   * @param  {Function} callback
+   * @param  {Object} [context]
+   * @param  [...more args...] can be provided and will be passed to the callback}
+   */
+  after: function(topic, callback, context /*adtl args*/) {
+    return this._registerEvent('after', topic, callback, context, Array.prototype.slice.call(arguments, 3));
+  },
+
+  // once: function(topic, callback, context /*adtl args*/) {
+  //   return this._registerEvent('once', topic, callback, context, Array.prototype.slice.call(arguments, 3));
+  // },
+
+  _registerEvent: function(type, topic, callback, context, adtlArgs) {
+    var listener = {
+      callback: callback,
+      context: context,
+      adtlArgs: adtlArgs
+    };
+
+    if (!this.__topics[topic]) {
+      this.__topics[topic] = {};
+    }
+
+    this.__topics[topic][type] = this.__topics[topic][type] || [];
+
+    this.__topics[topic][type].push(listener);
+
+  },
+
+  /**
+   * Fire an event on this object.
+   * @method fire
+   * @param  {String} topic
+   * @param  {Function} defaultFn a function that will be 
+   *   executed unless the default function was cancelled by a listener
+   * @param  {Object} adtlProperties will be added to the event object
+   */
+  fire: function(topic, defaultFn, adtlProperties) {
+    var eventFacade = new Event(topic, this);
+    if (Lang.isObject(adtlProperties)) {
+      ObjectUtils.mix(eventFacade, adtlProperties);
+    }
+
+    EventTarget._dispatch(this, 'on', topic, eventFacade);
+    // EventTarget._dispatch('once', topic, eventFacade, true);
+
+    if (!eventFacade.defaultPrevented) {
+      if (Lang.isFunction(defaultFn)) {
+        defaultFn();
+      }
+
+      EventTarget._dispatch(this, 'after', topic, eventFacade);
+    }
+
+  },
+
+  /**
+   * Bubble events up from this to eventTarget
+   * eventTarget will receive all events of this.
+   * 
+   * @method addBubbleTarget
+   */
+  addBubbleTarget: function(eventTarget) {
+    if (!this.isEventTarget) {
+      throw new Error('Can only bubble to classes that agument EventTarget');
+    }
+    this.__bubbleTargets.push(eventTarget);
+  }
+
+  //NOTE: More mechanisms may be needed to prevent memory leaks. 
```

Comment:
Classic problem of the Subject holding onto the last reference to something that was assumed destroyed. Mechanisms may include ways for listeners to detach on destruction (but of course js has no destructors)